agent/tests/test_php_wrapper.c

```diff
@@ -85,6 +90,460 @@ NR_PHP_WRAPPER(test_add_2_arrays) {
   NR_PHP_WRAPPER_CALL;
 }
 NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(test_name_txn_before_not_ok) {
+  nr_txn_set_path("UnitTest", NRPRG(txn), wraprec->funcname,
+                  NR_PATH_TYPE_ACTION, NR_NOT_OK_TO_OVERWRITE);
+
+  NR_PHP_WRAPPER_CALL;
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(test_name_txn_before_ok) {
+  nr_txn_set_path("UnitTest", NRPRG(txn), wraprec->funcname,
+                  NR_PATH_TYPE_ACTION, NR_OK_TO_OVERWRITE);
+
+  NR_PHP_WRAPPER_CALL;
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(test_name_txn_after_not_ok) {
+  NR_PHP_WRAPPER_CALL;
+  nr_txn_set_path("UnitTest", NRPRG(txn), wraprec->funcname,
+                  NR_PATH_TYPE_ACTION, NR_NOT_OK_TO_OVERWRITE);
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(test_name_txn_after_ok) {
+  NR_PHP_WRAPPER_CALL;
+  nr_txn_set_path("UnitTest", NRPRG(txn), wraprec->funcname,
+                  NR_PATH_TYPE_ACTION, NR_OK_TO_OVERWRITE);
+}
+NR_PHP_WRAPPER_END
+
+static void populate_functions() {
+  tlib_php_request_eval("function three($a) { return $a; }" TSRMLS_CC);
+  tlib_php_request_eval("function two($a) { return three($a); }" TSRMLS_CC);
+  tlib_php_request_eval("function one($a) { return two($a); }" TSRMLS_CC);
+}
+/*
+ * This function is meant to wrap/test when only ONE before/after special
+ * callback is chosen for one, two, and three. If one_before is configured,
+ * one_after must be NULL.
+ */
+static void setup_nested_framework_calls(nrspecialfn_t one_before,
```

Comment:
This function does a little more than just 'setup'. In fact it does execute the test. Consider `execute_nested_calls_scenario` or `test_txn_naming_in_nested_calls`.

Reply:
please see: [857fa16](https://github.com/newrelic/newrelic-php-agent/pull/615/commits/857fa1642e98a49e0a3b46e1404af1b6a4146637)